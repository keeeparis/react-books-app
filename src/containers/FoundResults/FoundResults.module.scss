.wrapper {
  padding: 10px 20px;
  background-color: var(--color-1);
  cursor: default;
  box-shadow: 1px 6px 20px 1px rgb(0 0 0 / 14%);
  border-radius: 8px;
  min-width: 200px;
  text-align: center;
}
