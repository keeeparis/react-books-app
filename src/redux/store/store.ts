import { configureStore } from '@reduxjs/toolkit'
import { setupListeners } from '@reduxjs/toolkit/dist/query'

import mainReducer from '../mainFeature/mainSlice'

import { booksApi } from '../services/books'

export const store = configureStore({
  reducer: {
    main: mainReducer,
    [booksApi.reducerPath]: booksApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(booksApi.middleware),
})

// setupListeners(store.dispatch)

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
