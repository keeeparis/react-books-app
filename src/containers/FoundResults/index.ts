import FoundResults from './FoundResults'

export default FoundResults
