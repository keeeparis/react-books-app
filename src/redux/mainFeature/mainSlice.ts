import { createSlice } from '@reduxjs/toolkit'
import { RootState } from '../store/store'

interface mainState {
  page: number
  skip: boolean
  input: string
}

const initialState: mainState = {
  page: 1,
  skip: true,
  input: '',
}

export const mainSlice = createSlice({
  name: 'main',
  initialState,
  reducers: {
    updatePage: (state, action) => {
      state.page += action.payload
    },
    updateSkip: (state, action) => {
      state.skip = action.payload
    },
    updateInput: (state, action) => {
      state.input = action.payload
    },
  },
})

export const { updatePage, updateSkip, updateInput } = mainSlice.actions

export default mainSlice.reducer

export const selectPage = (state: RootState) => state.main.page
export const selectSkip = (state: RootState) => state.main.skip
export const selectInput = (state: RootState) => state.main.input
