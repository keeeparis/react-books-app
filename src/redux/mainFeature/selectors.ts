import { createSelector } from '@reduxjs/toolkit'

import { BOOKS_PER_PAGE } from '../../pages/App'
import { RootState } from '../store/store'

/* Select */
export const selectStartIndex = (state: RootState) => state.main.startIndex
export const selectSkip = (state: RootState) => state.main.skip
export const selectInput = (state: RootState) => state.main.input
export const selectAllBooks = (state: RootState) => state.main.books
export const selectCategory = (state: RootState) => state.main.category
export const selectSorting = (state: RootState) => state.main.sorting
export const selectTotalItems = (state: RootState) => state.main.totalItems

export const selectIsMoreResults = (state: RootState) =>
  state.main.totalItems > state.main.startIndex + BOOKS_PER_PAGE

/* Create Selectors */
export const selectIsAnyBooks = createSelector(selectAllBooks, (books) =>
  books ? !!books.length : 0
)

export const selectBookById = createSelector(
  [selectAllBooks, (state, id: string) => id],
  (books, id) => {
    return books.filter((book) => book.id === id)[0]
  }
)
