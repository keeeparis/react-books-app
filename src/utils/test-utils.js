import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import 'regenerator-runtime/runtime'
import { Provider } from 'react-redux'
import { store } from '../redux/store/store'
import { MemoryRouter } from 'react-router-dom'
import '../i18n/config'

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  )
}

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { userEvent }
export * from 'regenerator-runtime/runtime'
export * from '../i18n/config'

export { customRender as render }
