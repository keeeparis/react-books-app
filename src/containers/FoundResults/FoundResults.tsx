import React from 'react'
import { useTranslation } from 'react-i18next'

import styles from './FoundResults.module.scss'

import { useAppSelector } from '../../redux/hooks/hooks'
import { selectTotalItems } from '../../redux/mainFeature/selectors'
import Fade from '../Fade'

const FoundResults = () => {
  const { t } = useTranslation()
  const resultsNumber = useAppSelector(selectTotalItems)

  return (
    <>
      {!!resultsNumber && (
        <Fade>
          <div className={styles.wrapper}>
            {t('resultsWithCount', { count: resultsNumber })}
          </div>
        </Fade>
      )}
    </>
  )
}

export default FoundResults
