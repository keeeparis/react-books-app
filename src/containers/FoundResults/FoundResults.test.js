import { act } from 'react-test-renderer'
import { render, screen, fireEvent, userEvent } from 'test-utils'
import { updateTotalItemsResponse } from '../../redux/mainFeature/mainSlice'
import { store } from '../../redux/store/store'
import FoundResults from './FoundResults'

it('test FoundResults Component', () => {
  const { asFragment } = render(<FoundResults />)

  expect(asFragment()).toMatchSnapshot()

  act(() => {
    store.dispatch(updateTotalItemsResponse(200))
  })

  expect(screen.getByText(/200/)).toBeInTheDocument()

  expect(asFragment()).toMatchSnapshot()
})
