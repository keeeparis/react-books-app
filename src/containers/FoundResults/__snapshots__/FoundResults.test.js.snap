// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test FoundResults Component 1`] = `<DocumentFragment />`;

exports[`test FoundResults Component 2`] = `
<DocumentFragment>
  <div
    class="wrapper fade-appear fade-appear-active"
  >
    Found 200 results
  </div>
</DocumentFragment>
`;
